client/modules/IDE/pages/IDEView.jsx

```diff
@@ -77,265 +79,243 @@ function WarnIfUnsavedChanges() {
-        {this.props.ide.newFolderModalVisible && <NewFolderModal />}
-        {this.props.ide.uploadFileModalVisible && <UploadFileModal />}
-        {this.props.location.pathname === '/about' && (
-          <Overlay
-            title={this.props.t('About.Title')}
-            previousPath={this.props.ide.previousPath}
-            ariaLabel={this.props.t('About.Title')}
-          >
-            <About previousPath={this.props.ide.previousPath} />
-          </Overlay>
-        )}
-        {this.props.location.pathname === '/feedback' && (
-          <Overlay
-            title={this.props.t('IDEView.SubmitFeedback')}
-            previousPath={this.props.ide.previousPath}
-            ariaLabel={this.props.t('IDEView.SubmitFeedbackARIA')}
-          >
-            <Feedback />
-          </Overlay>
-        )}
-        {this.props.location.pathname.match(/add-to-collection$/) && (
-          <Overlay
-            ariaLabel={this.props.t('IDEView.AddCollectionARIA')}
-            title={this.props.t('IDEView.AddCollectionTitle')}
-            previousPath={this.props.ide.previousPath}
-            actions={<CollectionSearchbar />}
-            isFixedHeight
-          >
-            <AddToCollectionList
-              projectId={this.props.params.project_id}
-              username={this.props.params.username}
-              user={this.props.user}
-            />
-          </Overlay>
-        )}
-        {this.props.ide.shareModalVisible && (
-          <Overlay
-            title={this.props.t('IDEView.ShareTitle')}
-            ariaLabel={this.props.t('IDEView.ShareARIA')}
-            closeOverlay={this.props.closeShareModal}
-          >
-            <ShareModal
-              projectId={this.props.ide.shareModalProjectId}
-              projectName={this.props.ide.shareModalProjectName}
-              ownerUsername={this.props.ide.shareModalProjectUsername}
-            />
-          </Overlay>
-        )}
-        {this.props.ide.keyboardShortcutVisible && (
-          <Overlay
-            title={this.props.t('KeyboardShortcuts.Title')}
-            ariaLabel={this.props.t('KeyboardShortcuts.Title')}
-            closeOverlay={this.props.closeKeyboardShortcutModal}
-          >
-            <KeyboardShortcutModal />
-          </Overlay>
-        )}
-        {this.props.ide.errorType && (
-          <Overlay
-            title={this.props.t('Common.Error')}
-            ariaLabel={this.props.t('Common.ErrorARIA')}
-            closeOverlay={this.props.hideErrorModal}
-          >
-            <ErrorModal
-              type={this.props.ide.errorType}
-              closeModal={this.props.hideErrorModal}
-            />
-          </Overlay>
-        )}
-      </RootPage>
-    );
-  }
-}
+                  <Console />
+                </SplitPane>
+              </PreviewWrapper>
+              <EditorSidebarWrapper show={!ide.isPlaying}>
+                <FileDrawer show={ide.sidebarIsExpanded}>
```

Comment:
I think it's weird that we have all of these lines (284-310) in the `IDEView` component for the mobile sidebar, when the corresponding code for desktop is in the `Sidebar` component.  We should work on combining the two sidebar codes.

We want the the same dropdown options (add file, add folder, upload file) on both versions.  The ➕ button that you designed for the mobile looks really nice and we can use that on the desktop version too.

Reply:
yeah!